Runtime/Client/LootLockerServerRequest.cs

```diff
@@ -240,7 +240,11 @@ public static void CallDomainAuthAPI(string endPoint, LootLockerHTTPMethod httpM
         public static void UploadFile(string endPoint, LootLockerHTTPMethod httpMethod, byte[] file, string fileName = "file", string fileContentType = "text/plain", Dictionary<string, string> body = null, Action<LootLockerResponse> onComplete = null, bool useAuthToken = true, LootLocker.LootLockerEnums.LootLockerCallerRole callerRole = LootLocker.LootLockerEnums.LootLockerCallerRole.User)
         {
             Dictionary<string, string> headers = new Dictionary<string, string>();
-
+            if (file.Length == 0)
+            {
+                LootLockerLogger.GetForLogLevel(LootLockerLogger.LogLevel.Error)("File content is empty, not allowed.");
```

Comment:
Please invoke the callback with an error message as well. I've done this in the method above for inspiration (line 222).